tools/autograd/gen_autograd_functions.py

```diff
@@ -162,6 +162,8 @@ def save_arg(arg, is_output):
             saved_variables.append('std::vector<int64_t> {};'.format(name))
         elif arg['type'] == 'c10::optional<IntArrayRef>':
             saved_variables.append('c10::OptionalArray<int64_t> {};'.format(name))
+        elif arg['type'] == 'c10::optional<ArrayRef<double>> ':
```

Comment:
I will bet my job that none of you reviewers would have caught the extra space in this string literal, which makes this clause totally ineffective.  You know who caught it?  ASAN caught it.

#thanksASAN